### PR TITLE
[UnifiedTransformer] Optimize tokenizer and generation api

### PR DESCRIPTION
**Pull request by xiemoyuan**
### PR types
Function optimization

### PR changes
APIs

### Describe
1. Added the dialogue_encode in tokenizer
2. Fixed some bugs in generation API.
3. Added DuConv dataset.
4. Added example for UnifiedTransformer.

### CODE REVIEW

File: paddlenlp/transformers/unified_transformer/tokenizer.py [hunk 1]
```diff
@@ -417,3 +423,121 @@ def load_vocabulary(filepath,
...
+                        knowledge=None,
+                        task_type='chitchat',
+                        max_seq_len=512,
+                        max_response_len=128,
+                        max_knowledge_len=128,
+                        return_position_ids=True,
+                        return_token_type_ids=True,
+                        return_length=False):
+        """
+        Main method to encode the single-turn or mulyi-turn dialogue conversation.
```
Comment (ZeyuChen) [inline paddlenlp/transformers/unified_transformer/tokenizer.py]: `mulyi-turn` -> `multi-turn`

File: paddlenlp/transformers/unified_transformer/tokenizer.py [hunk 2]
```diff
@@ -417,3 +423,121 @@ def load_vocabulary(filepath,
...
+                        task_type='chitchat',
+                        max_seq_len=512,
+                        max_response_len=128,
+                        max_knowledge_len=128,
+                        return_position_ids=True,
+                        return_token_type_ids=True,
+                        return_length=False):
+        """
+        Main method to encode the single-turn or mulyi-turn dialogue conversation. 
+        It will returns a dictionary containing the encoded sequence and other
```
Comment (ZeyuChen) [inline paddlenlp/transformers/unified_transformer/tokenizer.py]: will returns -> will return

**Review by ZeyuChen — COMMENTED** [review]
Fix few typos.

File: paddlenlp/transformers/unified_transformer/tokenizer.py [hunk 3]
```diff
@@ -58,6 +58,12 @@ class UnifiedTransformerTokenizer(PretrainedTokenizer):
         },
     }
 
+    TASK_TYPE_TO_SPECIAL_TOKEN = {
+        "chitchat": "[CHAT]",
+        "knowledge": "[KNOW]",
+        "recommend": "[RECO]"
+    }
```
Comment (guoshengCS) [inline paddlenlp/transformers/unified_transformer/tokenizer.py]: 这几个special token要不也和unk、pad一样作为`__init__`和`load_vocabulary`的命名关键字参数来设置和获取

File: paddlenlp/transformers/unified_transformer/tokenizer.py (hunk 1 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/tokenizer.py (hunk 2 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: Done. Thanks.

File: paddlenlp/transformers/unified_transformer/tokenizer.py (hunk 3 above)
Comment (xiemoyuan) [reply paddlenlp/transformers/unified_transformer/tokenizer.py]: Done. thanks.

**Pull request closed without merging**